napari/_qt/qt_color_dialog.py

```diff
@@ -0,0 +1,312 @@
+        if initial_color is not None:
+            self.setColor(initial_color)
+
+        layout.addWidget(self.color_swatch)
+        layout.addWidget(self.line_edit)
+
+    @property
+    def color(self):
+        """Return the current color."""
+        return self.color_swatch.color
+
+    def _on_line_edit_edited(self):
+        """When the user hits enter or loses focus on the LineEdit widget."""
+        text = self.line_edit.text()
+        rgb_match = rgba_regex.match(text)
+        if rgb_match:
+            text = [float(x) for x in rgb_match.groups() if x]
+        self.color_swatch.setColor(text)
+
+    @Slot(np.ndarray)
+    def _on_swatch_changed(self, color: np.ndarray):
+        """Receive QColorSwatch change event, update the lineEdit, re-emit."""
+        self.line_edit.setText(color)
+        self.color_changed.emit(color)
+
+
+class QColorSwatch(QFrame):
+    """A QFrame that displays a color and can be clicked to show a QColorPopup.
+
+    Parameters
+    ----------
+    parent : QWidget, optional
+        parent widget, by default None
+    tooltip : Optional[str], optional
+        Tooltip when hovering on swatch,
+        by default 'click to set color'
+    initial_color : ColorType, optional
+        initial color, by default will be transparent
+
+    Attributes
+    ----------
+    color : np.ndarray
+        The current color
+
+    Signals
+    -------
+    color_changed : np.ndarray
+        Emits the new color when the current color changes.
+    """
+
+    color_changed = Signal(np.ndarray)
+
+    def __init__(
+        self,
+        parent: Optional[QWidget] = None,
+        tooltip: Optional[str] = None,
+        initial_color: Optional[ColorType] = None,
+    ):
+        super().__init__(parent)
+        self.setObjectName('colorSwatch')
+        self.setToolTip(tooltip or 'click to set color')
+        self.setCursor(Qt.PointingHandCursor)
+
+        self.color_changed.connect(self._update_swatch_style)
+        self._color: np.ndarray = TRANSPARENT
+        if initial_color is not None:
+            self.setColor(initial_color)
+
+    @property
+    def color(self):
+        """Return the current color"""
+        return self._color
+
+    @Slot(np.ndarray)
+    def _update_swatch_style(self, color: np.ndarray) -> None:
+        """Convert the current color to rgba() string and update appearance."""
+        rgba = f'rgba({",".join(map(lambda x: str(int(x*255)), self._color))})'
+        self.setStyleSheet('#colorSwatch {background-color: ' + rgba + ';}')
+
+    def mouseReleaseEvent(self, event: QEvent):
+        """Show QColorPopup picker when the user clicks on the swatch."""
+        if event.button() == Qt.LeftButton:
+            initial = QColor(*(255 * self._color).astype('int'))
+            popup = QColorPopup(self, initial)
+            popup.colorSelected.connect(self.setColor)
+            popup.show_right_of_mouse()
+
+    def setColor(self, color: AnyColorType) -> None:
```

Comment:
`@color.setter`?

Reply:
please no, much hard to use to connect to signals.  i know we don't have to do everything 
the Qt way, but I think getting too far away from the standard qt patterns and conventions starts to just confuse anyone who _is_ familiar with it.

Reply:
Oh ok! Sorry, "anyone who is familiar with it" and myself is an empty set. =P The question mark was genuine! =)

Reply:
lol.  ok :)  yeah... just cause it's funny, here's what using a `@setter` would do to line 179 just above it:
```python
popup.colorSelected.connect(self.setColor)

popup.colorSelected.connect(lambda color: self.__class__.color.fset(self, color))
```